R/01-qc.R

```diff
@@ -1,39 +1,102 @@
 
-#' This is a title for a function
+#' A function to run QC
 #' @description This is a function here's where we describe what it does
 #' @param parameter Here's a parameter let's describe it here
 #' @export
 #' @importFrom tidyr pivot_longer
-#' @import ggplot2
+#' @importFrom magrittr %>%
+#' @import ggplot2 pheatmap
 #' @examples \dontrun{
 #'
 #' }
 
-run_qc <- function(gimap_data, plots_dir = "./qc_plots", wide_ar = 0.75, square_ar = 1) {
 
-  if (!dir.exists(plots_dir)) {
+run_qc <- function(gimap_dataset, plasmid_cutoff = NULL, use_combined = TRUE,
+                   params = list(cell_line = NULL), plots_dir = "./qc_plots", overwrite = FALSE, output_format = NULL, output_file = "QC_Report", output_dir = "./",
+                   wide_ar = 0.75, square_ar = 1) {
+  
+  if (!dir.exists(plots_dir)){
     dir.create(plots_dir, showWarnings = TRUE)
   }
+  
+  if (!("gimap_dataset" %in% class(gimap_dataset))) stop("This function only works with gimap_dataset objects which can be made with the setup_data() function.")
+  
+  # for Rmd report
+  list_of_qc_things <- list(counts_cdf = NULL,
+                            sample_cpm_histogram = NULL,
+                            sample_cor_heatmap_unfiltered = NULL,
+                            zerocount_df = NULL,
+                            plasmid_hist_nocutoff = NULL,
+                            plasmid_stats = NULL,
+                            plasmid_hist_cutoff = NULL,
+                            plasmid_filter_report = NULL,
+                            combined_filter_df = NULL,
+                            which_filter_df = NULL
+                            )
+  
+  counts_cdf <- qc_cdf(gimap_dataset, wide_ar = wide_ar)
+  
+  list_of_qc_things$counts_cdf <- counts_cdf #put it in the report
+  save_plot(counts_cdf, params = params, out_dir = plots_dir)
```

Comment:
One saves the plot as a png/pdf and the other saves the plot in a list that I pass to the template Rmd/html report. I think for publication purposes it's a good setup to make it available in multiple ways in case you want just one specific plot to be included in the manuscript, but then you can include the QC report as like a supplemental file or just use it internally. I was even considering trying to make it so the report lists the actual file name/where the plot is saved.

Reply:
Ok, I'll incorporate listing where the files are saved

Reply:
This has been handled in [commit f08e240](https://github.com/FredHutch/gimap/commit/f08e240bedb0326205bf9cec74c88e5508859026)